src/Symfony/Component/Serializer/Mapping/Factory/CacheClassMetadataFactory.php

```diff
@@ -44,8 +44,8 @@ public function __construct(ClassMetadataFactoryInterface $decorated, CacheItemP
     public function getMetadataFor($value)
     {
         $class = $this->getClass($value);
-        // Key cannot contain backslashes according to PSR-6
-        $key = strtr($class, '\\', '_');
+        // Key cannot contain backslashes, colons, brackets, slashes and @ according to PSR-6
+        $key = false !== strpbrk($class, '\{}()/@:') ? rawurlencode($class) : $class;
```

Comment:
one last note: let's optimise for the most likely case: the backslash is going to be here 99% of the time.
`$key = rawurlencode($class);` should be the best here.

Reply:
@nicolas-grekas Thanks. I applied suggested change.